### ISSUE TITLE
Clicking on a Friend will not show the friend on the map until the next location push request (HTTP)

### ISSUE THREAD

**Issue by sizemoreap** [owntracks/android#1382]
If I go to my Friends list and click on a name, nothing happens until my location has changed enough or the time elaspsed to trigger a HTTP request. Thus, I'm forced to click on my friend's name and then click the button in the top-right of the header bar to get the map to scroll to the location of my friend.

Can a developer add an extra line of code behind the click event on the Friend's list to force an update so the map will scroll right then and there?

Thanks!

**Comment by growse** [comment]
Hmm, this doesn't work for you?

![device-2023-03-10-173021](https://user-images.githubusercontent.com/378067/224383558-1cee6155-78f9-408d-ad20-7e741db72594.gif)

**Comment by sizemoreap** [comment]
No, it does not. Until I've moved enough or the timeout has occurred to auto push an update to me server, it won't auto scroll to the friend I've selected unless I push the button in the top right to force an update. I'm using the OSS version from one of the Droidify repos, btw.

**Comment by sizemoreap** [comment]
Probably also relevant; I apologize that I didn't think of this before. I'm running GrapheneOS with Google Play Services sandboxed so I have a persistent notification to keep this running constantly.

**Comment by growse** [comment]
> No, it does not. Until I've moved enough or the timeout has occurred to auto push an update to me server, it won't auto scroll to the friend I've selected unless I push the button in the top right to force an update. I'm using the OSS version from one of the Droidify repos, btw.

Does it just go back to where you were on the map?

**Comment by sizemoreap** [comment]
Yeah. So if I click on a friend, the Friend activity goes away and I'm back to looking at my own initials on the map.

**Comment by sizemoreap** [comment]
Is there any more info I can provide that will help? 

GrapheneOS (latest build)
Google Play Services sandboxed
Downloaded OwnTracks from Izzy F-Droid repo

**Comment by growse** [comment]
Do you get the popup card for the new contact at the bottom of the map?

![image](https://user-images.githubusercontent.com/378067/226920684-14acfbb6-f4a7-4287-805d-08bdad8e0960.png)

**Comment by sizemoreap** [comment]
Yes, the pop up card does appear for the contact I select. The map just doesn't move to their location and just shows mine until I force an update.

**Comment by sizemoreap** [comment]
I understand. Thanks for looking into it at any rate.

May try to study up on Kotlin and see if I can figure it out.

**Comment by growse** [comment]
> I am also Grapheneos (without Google-Play-Services), and I experience the exact same Behaviour. Everything else works fine.

Could I ask you to try and reproduce on the next release (when it eventually comes out)? I'm still having trouble reproducing this.

> Also: when I tap on the Friend's Popup-Card, then on the Submenu (three Dots in the lower Left), then on “navigate” – the Submenu disappears, but nothing else happens, the Map does not scroll to the Friend, what I suppose should happen.

Just checked the code: "Navigate" has historically meant "Open Google Maps and start navigation to this location" by firing an intent to `"google.navigation:q=$latitude,$longitude"`. Obviously on a non-Google device, this is going to not work, so we should probably think of a better / more generic approach.

**Comment by cocker-cc** [comment]
> Could I ask you to try and reproduce on the next release (when it eventually comes out)?  

Yes, for sure.  Thanks to all of you for this great App 👍

**Comment by ericje627** [comment]
I would like to comment on this issue as I'm experiencing similar a behaviour that clicking on a friend from the menu does not move the map to where the friend's marker is. The marker is displayed when navigating manually to its location. I'm running the Google Play version (v. 2.4.10) and this issue seems to arise only when using the OpenStreetmap map layer. The default Google Map layers (either 'Google Default', 'Google Hybrid' or 'Google Terrain') all work fine.

Like mentioned in https://github.com/owntracks/android/issues/1382#issuecomment-1559631374, tap and hold on the bar at the bottom of the screen _does_ center the OSM layer to the friend's location.

Steps to reproduce:
- Select 'OpenStreetMap' as Map layer in the map view
- Open 'Friends' from the menu and click on one of the entries
- The friend's information is displayed in the bar at the bottom of the screen, yet the map stays at the current position
- (Workaround) Tap and hold on the bar at the bottom of the screen centers the map around the friend's marker.

Hope this helps. Let me know if there is anything that I can test from my side.

**Comment by kri164** [comment]
This bug remains on v.2.4.12. LineageOS, non google play version.

**Comment by growse** [comment]
2.4.11/12 we're just point releases on 2.4.10 to appease the play store gods. There's no bugfixes or new features.

New stuff is coming in 2.5.
